File: src/app/services/question.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  private apiUrl = 'http://localhost:8080/api/questions';

  constructor(private http: HttpClient) { }

  // Metodo per ottenere tutte le categorie disponibili
  getAllCategories(): Observable<string[]> {
    return this.http.get<string[]>(`${this.apiUrl}/categories`);
  }

  // Metodo per ottenere una domanda casuale in base alla categoria
  getQuestionByCategory(category: string): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/random?category=${category}`);
  }
}

File: TriviaBackEnd/TriviaDuel/src/main/java/com/Capstone/TriviaDuel/Service/QuestionService.java
package com.Capstone.TriviaDuel.Service;




import com.Capstone.TriviaDuel.Model.Question;
import com.Capstone.TriviaDuel.Repository.QuestionRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Random;
import java.util.stream.Collectors;

@Service
public class QuestionService {
    @Autowired
    private QuestionRepository questionRepository;


    // Metodo per ottenere una domanda casuale in base all'argomento
    public Question getRandomQuestionByCategory(String category) {
        List<Question> questions = questionRepository.findByCategory(category);
        if (questions.isEmpty()) {
            return null;
        }
        Random random = new Random();
        return questions.get(random.nextInt(questions.size()));
    }
    // Metodo per ottenere tutte le categorie
    public List<String> getAllCategories() {
        return questionRepository.findAll()
                .stream()
                .map(Question::getCategory)
                .distinct()
                .collect(Collectors.toList());
    }

    // Metodo per creare una nuova domanda
    public Question createQuestion(Question question) {
        return questionRepository.save(question);
    }
}

